source/common/upstream/outlier_detection_impl.cc

```diff
@@ -248,6 +248,13 @@ void DetectorImpl::onConsecutive5xxWorker(HostSharedPtr host) {
     return;
   }
 
+  // There are two outcomes here. The ejection will be enforced,
```

Comment:
the enforcement decision happens in the `ejectHost` function, so at this point the ejection can still be enforced or not.

If we do the reset after enforcing the ejection, we leave hosts whose ejection was not enforced in the problematic state.